caas/kubernetes/provider/k8s.go

```diff
@@ -807,6 +807,32 @@ func (k *kubernetesClient) Service(appName string) (*caas.Service, error) {
 			Scope: network.ScopePublic,
 		})
 	}
+
+	deploymentName := k.deploymentName(appName)
+	statefulsets := k.AppsV1().StatefulSets(k.namespace)
+	ss, err := statefulsets.Get(deploymentName, v1.GetOptions{})
+	if err == nil {
+		scale := 0
+		if ss.Spec.Replicas != nil {
+			scale = int(*ss.Spec.Replicas)
+		}
+		result.Scale = scale
+		return &result, nil
+	}
+	if !k8serrors.IsNotFound(err) {
+		return nil, errors.Trace(err)
+	}
+
+	deployments := k.AppsV1().Deployments(k.namespace)
+	deployment, err := deployments.Get(deploymentName, v1.GetOptions{})
+	if err != nil {
```

Comment:
should we translate `k8serrors.NotFound` to `errors.NotFound` here?

Reply:
theoretically should not get that error but we could so have added extra check